File: backend/src/main/java/vn/edu/fpt/doghandbook/backend/service/TrainingSpecialtyService.java
package vn.edu.fpt.doghandbook.backend.service;

import vn.edu.fpt.doghandbook.backend.dto.request.TrainingSpecialtyRequest;
import vn.edu.fpt.doghandbook.backend.dto.response.PageResponse;
import vn.edu.fpt.doghandbook.backend.dto.response.TrainingSpecialtyResponse;

public interface TrainingSpecialtyService {

    PageResponse<TrainingSpecialtyResponse> getAll(int page, int size, String search);

    TrainingSpecialtyResponse getById(Integer id);

    TrainingSpecialtyResponse create(TrainingSpecialtyRequest request);

    TrainingSpecialtyResponse update(Integer id, TrainingSpecialtyRequest request);

    void delete(Integer id);
}

File: web-admin/src/services/trainingSpecialtyService.js
import api from './api';

const buildQuery = (page = 0, size = 20, search = '') => {
  const params = new URLSearchParams();
  params.append('page', String(page));
  params.append('size', String(size));
  if (search?.trim()) {
    params.append('search', search.trim());
  }
  return params.toString();
};

export const trainingSpecialtyService = {
  getAll: (page = 0, size = 20, search = '') =>
    api.get(`/training-specialties?${buildQuery(page, size, search)}`),
  getById: (id) => api.get(`/training-specialties/${id}`),
  create: (data) => api.post('/training-specialties', data),
  update: (id, data) => api.put(`/training-specialties/${id}`, data),
  delete: (id) => api.delete(`/training-specialties/${id}`),
};

export default trainingSpecialtyService;
